ui/components/app/assets/nfts/nft-details/nft-full-image.tsx

```diff
@@ -24,9 +24,11 @@ import { useI18nContext } from '../../../../../hooks/useI18nContext';
 import { ASSET_ROUTE } from '../../../../../helpers/constants/routes';
 import useGetAssetImageUrl from '../../../../../hooks/useGetAssetImageUrl';
 
+type Params = { asset: string; id: string };
+
 export default function NftFullImage() {
   const t = useI18nContext();
-  const { asset, id } = useParams<{ asset: string; id: string }>();
+  const { asset, id } = useParams<Params>() as Params;
```

Comment:
This cast does not appear to be necessary. It's also generally discouraged by our TypeScript guidelines (unless there's no way to resolve the type error using a type guard, annotation, or correct types - [see here for details](https://github.com/MetaMask/contributor-docs/blob/main/docs/typescript.md#avoid-type-assertions-by-using-type-guards-to-improve-type-inference))

```suggestion
  const { asset, id } = useParams<Params>();
```

Reply:
Sorry, I see the type error this causes now. I missed it at first when trying this locally. Investigating...

Reply:
Ah I see.... we're not handling the undefined case. Hmm.

Reply:
This type error seems like a legitimate bug; the component assumes the parameter exists, but it might not exist. Using a type cast hides this problem. But it also doesn't seem appropriate to fix it in this PR, since this PR is focused on the React Router upgrade.

Maybe we can replace the type casts with `// @ts-expect-error TODO: Fix this type error by handling undefined parameters`. Thoughts?

Reply:
Got rid of the cast and added `// @ts-expect-error TODO: Fix this type error by handling undefined parameters`